rest/importtest/collections_import_test.go

```diff
@@ -410,6 +429,100 @@ func TestMultiCollectionImportRemoveCollection(t *testing.T) {
 	// there should be 10 documents datastore1_{10..19}
 	require.Equal(t, int64(docCount), rt.GetDatabase().DbStats.SharedBucketImport().ImportCount.Value())
 
+	// Get the import version from the persisted config to ensure it didn't change
+	updatedImportVersion := GetImportVersion(t, rt, dbName)
+	require.Equal(t, initialImportVersion, updatedImportVersion)
+
+}
+
+// TestImportVersionWriteVariations - ensure import version is updated for all different APIs that can modify the collection set
+func TestImportVersionWriteVariations(t *testing.T) {
+
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyDCP)
+	base.SkipImportTestsIfNotEnabled(t)
+	numCollections := 3
+	base.RequireNumTestDataStores(t, numCollections)
+
+	ctx := base.TestCtx(t)
+	testBucket := base.GetTestBucket(t)
+	defer testBucket.Close(ctx)
+
+	rtConfig := &rest.RestTesterConfig{
+		CustomTestBucket: testBucket.NoCloseClone(),
+		PersistentConfig: true,
+	}
+
+	rt := rest.NewRestTester(t, rtConfig)
+	defer rt.Close()
+
+	// Set up collection payloads for 1, 2, 3 collections
+	oneCollectionConfig := rest.GetCollectionsConfig(t, testBucket, 1)
+	oneCollectionConfigJSON, err := json.Marshal(oneCollectionConfig)
+	require.NoError(t, err)
```

Comment:
fyi about testing helper functions for readability

```suggestion
	oneCollectionConfigJSON := base.MustJSONMarshal(oneCollectionConfig)
```

and we can avoid the extra variable, or even convert this just as we are ready to make a rest call.